src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -152,7 +200,9 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
             var encoding = defaultEncoding ?? Encoding.UTF8;
             var sourceFileInfoTasks = sourceDocuments.Select(sd => _pdbSourceDocumentLoaderService.LoadSourceDocumentAsync(tempFilePath, sd, encoding, telemetry, useExtendedTimeout, cancellationToken)).ToArray();
             var sourceFileInfos = await Task.WhenAll(sourceFileInfoTasks).ConfigureAwait(false);
-            if (sourceFileInfos is null || sourceFileInfos.Where(t => t is null).Any())
+
+            // No point continuing if no source files were found
+            if (!sourceFileInfos.WhereNotNull().Any())
```

Comment:
>Any() [](http://example.com/codeflow?start=48&length=5)

`Any(info => info is null)` should also do

Reply:
Ended up reverting this anyway, will revisit later